kernel/src/syscall.rs

```diff
@@ -516,117 +563,200 @@ impl SyscallReturn {
             SyscallReturn::AllowReadOnlyFailure(_, _, _) => false,
             SyscallReturn::SubscribeFailure(_, _, _) => false,
             SyscallReturn::YieldWaitFor(_, _, _) => true,
+            SyscallReturn::SuccessUsize(_) => true,
         }
     }
 
     /// Encode the system call return value into 4 registers, following the
     /// encoding specified in TRD104. Architectures which do not follow TRD104
     /// are free to define their own encoding.
-    pub fn encode_syscall_return(&self, a0: &mut u32, a1: &mut u32, a2: &mut u32, a3: &mut u32) {
+    pub fn encode_syscall_return_32bit_trd104(
+        &self,
+        a0: &mut u32,
+        a1: &mut u32,
+        a2: &mut u32,
+        a3: &mut u32,
+    ) {
+        assert!(
+            core::mem::size_of::<CapabilityPtr>() == core::mem::size_of::<u32>()
+                && core::mem::align_of::<u32>() >= align_of::<CapabilityPtr>(),
+            "encode_syscall_return used on a 64-bit platform or CHERI platform"
+        );
```

Comment:
Can we validate that this is compiled away on 32-bit platforms? I think we don't want to introduce a panic if we don't have to as it significantly increases code size.

Reply:
It would be _highly_ surprising if `if [const exp false]` did not eliminate itself as dead code.

If this was liable to happen, all of the config that Tock has would fail and this has already been vetted.